### PR TITLE
Pull Request for Issue1601: Adding Amptek to new SGM

### PR DESCRIPTION
**Pull request by iainworkman**
- Added instances of an amptek detector to the SGM Beamline class
- Added a view for the amptek detector to the SGM App Controller

### CODE REVIEW

**Comment by iainworkman** [comment]
Need to sit down with the beam and make sure this is working as expected. @davidChevrier said he'd point out a few things he usually tests when working with the ampteks.

**Comment by iainworkman** [comment]
@dretrex, @davidChevrier suggested that you might be able to help make this really useful with respect to regions and maps. (this can be merged in pending review, any changes you think would be pertinent can be made in another issue).

File: source/application/SGM/SGMAppController.cpp [hunk 1]
```diff
@@ -83,6 +84,32 @@ void SGMAppController::shutdown() {
...
+		mw_->addPane(amptekDetectorView_, "Amptek Detector", "Amptek Detector", ":/system-software-update.png");
+		connect(amptekDetectorView_, SIGNAL(resized()), this, SLOT(onAmptekDetectorViewResized()));
+	}
+}
+
+void SGMAppController::onAmptekDetectorViewResized()
+{
+	// Hack solution copied over from the old SGM App Controller.
+	// If this timer is not here it doesn't always resize properly.
+	QTimer::singleShot(100, this, SLOT(resizeToMinimum()));
```
Comment (dretrex) [inline source/application/SGM/SGMAppController.cpp]: This is really weird.  Why doesn't it resize properly?

**Comment by dretrex** [comment]
Overall, looks good aside from the one question.  What do you mean by make it really useful?  I think I'm missing the context.
Time Estimate: 5 minutes

File: source/application/SGM/SGMAppController.cpp (hunk 1 above)
Comment (iainworkman) [inline source/application/SGM/SGMAppController.cpp]: Dunno, just copied it over. I could give it a try and see how it behaves without the timer. Now seems like as good a chance as any.

**Comment by iainworkman** [comment]
Well it's collecting spectra, and over a 2D map this means a 2D map of spectra, which can't be displayed. @davidChevrier said you'd have some suggestions on setting up RoIs to get it to actually produce workable data that can be viewed in AM.

File: source/application/SGM/SGMAppController.cpp (hunk 1 above)
Comment (dretrex) [inline source/application/SGM/SGMAppController.cpp]: Sounds good.  Let me know the results.

**Comment by dretrex** [comment]
Yeah, @davidChevrier told me about it this morning.  Come find me sometime and we can have a chat about it.

**Comment by dretrex** [comment]
Based on our discussion, I'm going to merge this and leave the new regions of interest issue to the new one.

**Pull request merged**